### PR TITLE
Fix some import bugs (mainly csv) plus better delete UX

### LINKED ISSUE #292: No contacts are imported from certain csv files
**Summary:**

There are three sample CSV files that can't be successfuly imported to our v4 contacts app.

The same files work on v3.

You may download them from issue https://proton.atlassian.net/browse/CLIENT-1781. Contact me if you don't have access,

**Example:**

image

### LINKED ISSUE #317: Deleting all contacts doesn't refresh the contact list
If you check the checkbox to select all the contacts in the list, and then delete them, the contact list is not updated, until refresh
1. Select all contacts
2. Click delete -> Contact list not updated

### PR DESCRIPTION
**Pull request by econdepe**
Bugs on import:
- Capturing csv's using underscore now (first_name -> first name)
- Better capture of several csv properties (mainly addresses, but also photo and logo, ignored before)
- Remove front-end timeout of 30s for the route submitting imported contacts. It seems the response is very slow these days from the back-end. I also reduced the import batch size.
- Improve speed of import by encrypting 5 contacts at a time.

Bugs on delete UX:
- After "delete all", we were not navigating to `/contacts` and we were leaving the checkbox for "check all" ticked.
- Only navigate away from current contact if it's deleted.
- `ContactsPlaceholder` was equal to `0` instead of `null` in some instances.

Fixes #292
Closes #317
Needs https://github.com/ProtonMail/proton-shared/pull/68 and https://github.com/ProtonMail/proton-shared/pull/69

### CODE REVIEW

File: src/app/components/import/ImportingModalContent.js [hunk 1]
```diff
@@ -119,11 +134,14 @@ const ImportingModalContent = ({ isVcf, file = '', vcardContacts, privateKey, on
 
             const indexMap = contacts.map(({ index }) => index);
 
+            const INFINITE_TIME = 1000 * 60 * 60; // 1 hour
```
Comment (EpokK) [inline src/app/components/import/ImportingModalContent.js]: It should be define at the top of this file.

File: src/app/constants.js [hunk 2]
```diff
@@ -55,4 +55,4 @@ export const API_SAFE_INTERVAL = 100; // API request limit: 100 requests / 10 se
 export const QUERY_EXPORT_MAX_PAGESIZE = 50; // in GET API route /contacts/export
 // Manual limit on number of imported contacts to be sent to the API,
 // so that the response does not take too long (there's a 30s limit)
-export const ADD_CONTACTS_MAX_SIZE = 500;
+export const ADD_CONTACTS_MAX_SIZE = 200;
```
Comment (EpokK) [inline src/app/constants.js]: How did you define this value?

File: src/app/components/import/ImportingModalContent.js (hunk 1 above)
Comment (econdepe) [reply src/app/components/import/ImportingModalContent.js]: Haha, ok. I wanted to be much more poetic here :grin:

File: src/app/constants.js (hunk 2 above)
Comment (econdepe) [reply src/app/constants.js]: Mmhh, yeah, I played with several values. Originally 500 was ok (in the sense that the API reply typically came well under 30s), but then API responses got much slower. I can bring the max size down to 100.

**Pull request merged**